pkgs/top-level/all-packages.nix

```diff
@@ -10398,6 +10398,12 @@ with pkgs;
 
   opencv = opencv4;
 
+  opencv-gtk = (opencv.override {
+    enableGtk3 = true;
+  }).overrideAttrs (finalAttrs: {
+    meta.platforms = lib.subtractLists lib.platforms.darwin finalAttrs.meta.platforms;
```

Comment:
Maybe a better way to handle this is to write `platforms = if enableGtk3 then lib.platforms.linux else lib.platforms.unix;` or `broken = enableGtk3 && stdenv.hostPlatform.isDarwin;` (definitely not comprehensive enough, but you get the idea) in `opencv4`.